dsl/interpreter.py

```diff
@@ -0,0 +1,164 @@
+        return None
+    if not os.path.isfile(input_name):
+        logger.error(f'File {input_name} does not exist')
+    file = open(input_name, "r")
+    lines = file.readlines()
+    # print whole file to a string with removing comments
+    is_commented = False
+    data = ''
+    # gr = Grammar()
+    # clean from comments
+    for line in lines:
+        line = line.strip()
+        ind_line_commented = len(line)
+        for i, c in enumerate(line):
+            if c == '@':
+                is_commented = not is_commented
+            if c == '#' and not is_commented and ind_line_commented == len(line):
+                ind_line_commented = i
+            if not (is_commented or ind_line_commented != len(line)):
+                data += c
+    # code = gr.parse(data)
+    code = simple_transform(data)
+    new_file_name = input_name[:len(input_name) - len(DSL_EXTENSION)] + '.py'
+    with open(new_file_name, 'w') as f:
+        f.write(code)
+    # close file
+    file.close()
+    return new_file_name
+
+def simple_transform(code='RETURN 0'):
+    python_code = 'import numpy as np\n'
+    code = code.replace('\n', ' \n ')
+    code = code.replace('{', ' {')
+    code = code.replace('}', ' }')
+    tokens = code.split(' ')
+    # print(code)
+    # print(tokens)
+    indent = 0
+    prefix = 0
+    if_flag = 0
+    else_flag = 0
+    for_flag = 0
+    for ind, token in enumerate(tokens):
+        if token == '':
+            continue
+        if token == '\n':
+            python_code += '\n' + ' '*(indent*4)
+            continue
+        if token == '{':
+            indent += 1
+            prefix += 1
+            if if_flag > 0:
+                if_flag -= 1
+                python_code += ' : '
+                continue
+            if else_flag > 0:
+                else_flag -= 1
+                python_code += ' : '
+                continue
+            if for_flag > 0:
+                for_flag -= 1
+                python_code += ' : '
+                continue
+
+            continue
+        if token == '}':
+            indent -= 1
+            prefix -= 1
+            if indent < 0 or prefix < 0:
+                logger.error(f'Wrong curly brackets')
+                return
+            continue
+        if token == 'IF':
+            python_code += 'if '
+            if_flag += 1
+            continue
+        if token == 'ELSE':
+            python_code += 'else'
+            else_flag += 1
+            continue
+        if token == 'FOREACH':
+            python_code += 'for '
+            for_flag += 1
+            continue
+        if token == 'IN':
+            python_code += 'in '
+            continue
+        if token == 'NUM' or token == 'BOOL':
+            continue
+        if token == 'SET':
+            if ind == len(tokens) - 1:
+                logger.error(f'Wrong set declaration')
+                return
+            name = tokens[ind + 1]
+            python_code += name + ' = np.array([])\n'
```

Comment:
numpy array is like C array: its size is fixed, hence such a creation is useless. It's better to use `list` first, do some operations on it, then, when its size is final, convert it to `np.array`